astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -640,3 +640,20 @@ def test_assignment_with_different_names(self):
         assert qra[1].value == np.array((-1000.0, -2000.0, -3000.0), qra.dtype)
         # Ensure we do not override dtype names of value.
         assert value.dtype.names == ("x", "y", "z")
+
+
+class TestArrayAPI:
+    def setup_method(self):
+        import warnings
+
+        warnings.filterwarnings("ignore", category=UserWarning)
```

Comment:
Should avoid global filterwarnings like this since this can affect other tests in the test suite. 

1. Why is this warning being ignored?
2. Can it be localized with a context manager on just the affected code?

Reply:
> Why is this warning being ignored?

importing array_api emits a warning about this being experimental.

>  Can it be localized with a context manager on just the affected code?

I tried using the pytest filterwarning fixture but I think it only works for `test_` functions?

I can make this whole class into a single functional test too, just wanted to follow the rest of the test suite in this file.

Reply:
I think applying `pytest.mark.filterwarnings` on the test class might work (not 100% sure).